torch_xla/csrc/tensor_impl.cpp

```diff
@@ -172,6 +175,35 @@ void XLATensorImpl::SetupSizeProperties() {
   }
 }
 
+void XLATensorImpl::SetupSymSizeProperties() {
+  size_t generation = tensor_->generation();
+  if (generation != generation_) {
+    // Fill up the basic dimension data members which the base class
+    // implementation uses in its APIs.
+    auto shape = tensor_->shape();
+    auto rank = tensor_->shape().get().rank();
+    c10::SmallVector<c10::SymInt, 5> sym_sizes;
+    numel_ = 1;
+    XLAIrBuilder a = XLAIrBuilder();
+    for (auto i : c10::irange(rank)) {
+      if (tensor_->shape().get().is_dynamic_dimension(i)) {
+        auto dim_node = a.MakeSizeNode(tensor_->GetIrValue(), i);
+        auto symint_node =
+            c10::make_intrusive<torch::lazy::SymIntNodeImpl>(dim_node);
+        auto sn = symint_node->toSymInt();
+        sym_sizes_.push_back(sn);
+        /*TODO(miladm): verify numel_ calculation after adding a dynamic op
+         */
+        numel_ *= dynamic_cast<SizeNode*>(dim_node.get())->getStaticValue();
```

Comment:
I've read the definition of numel_ in the base class but it's not clear to me what it means. Do you know the meaning of this member?

Reply:
number of elements

Reply:
Thanks